game/src/main/kotlin/world/gregs/voidps/world/interact/entity/npc/combat/type/GiantMole.kts

```diff
@@ -0,0 +1,212 @@
+import world.gregs.voidps.engine.entity.character.mode.move.exitArea
+import world.gregs.voidps.engine.entity.character.move.tele
+import world.gregs.voidps.engine.entity.character.npc.NPC
+import world.gregs.voidps.engine.entity.character.player.Player
+import world.gregs.voidps.engine.entity.character.player.Players
+import world.gregs.voidps.engine.entity.character.player.skill.Skill
+import world.gregs.voidps.engine.entity.character.setAnimation
+import world.gregs.voidps.engine.entity.obj.objectOperate
+import world.gregs.voidps.engine.entity.playerSpawn
+import world.gregs.voidps.engine.inject
+import world.gregs.voidps.engine.inv.inventory
+import world.gregs.voidps.engine.inv.itemChange
+import world.gregs.voidps.engine.inv.transact.operation.ReplaceItem.replace
+import world.gregs.voidps.engine.map.collision.random
+import world.gregs.voidps.type.Direction
+import world.gregs.voidps.type.Tile
+import world.gregs.voidps.world.interact.entity.combat.*
+import world.gregs.voidps.world.interact.entity.combat.hit.npcCombatHit
+import world.gregs.voidps.world.interact.entity.gfx.areaGraphic
+import world.gregs.voidps.world.interact.entity.player.equip.inventoryOption
+import world.gregs.voidps.world.interact.entity.sound.areaSound
+import kotlin.random.Random
+
+val logger = InlineLogger()
+val areas: AreaDefinitions by inject()
+val players: Players by inject()
+
+val acceptedTiles = listOf(
+    Tile(3005, 3376, 0),
+    Tile(2999, 3375, 0),
+    Tile(2996, 3377, 0),
+    Tile(2989, 3378, 0)
+)
+
+val giantMoleLair = areas["giant_mole_lair"]
+val gianMoleSpawns = areas["giant_mole_spawn_area"]
+val initialCaveTile: Tile = Tile(1752, 5237, 0)
+
+inventoryOption("Dig") {
+    val playerTile: Tile = player.tile
+    if(!acceptedTiles.contains(playerTile)) {
+        return@inventoryOption
+    }
+    player.setAnimation("dig_with_spade")
+    player.open("warning_dark")
+}
+
+objectOperate("Climb", "giant_mole_lair_escape_rope") {
+    player.setAnimation("climb_up")
+    player.tele(acceptedTiles.random())
+}
+
+interfaceOption(component = "proceed", id = "warning_dark") {
+    player.tele(initialCaveTile, clearInterfaces = true)
+}
+
+interfaceOption(component = "stayout", id = "warning_dark") {
+    player.closeInterfaces()
+}
+
+npcCombatHit("giant_mole") {
+    val currentHealth = it.levels.get(Skill.Constitution)
+    var shouldBurrow = false
+    if (it.fightStyle == "magic" && damage != 0) {
+        shouldBurrow = shouldBurrowAway(currentHealth)
+    } else if (it.fightStyle != "magic") {
+        shouldBurrow = shouldBurrowAway(currentHealth)
+    }
+    if (shouldBurrow && !it.hasClock("awaiting_mole_burrow_complete")) {
+        it.start("awaiting_mole_burrow_complete", 4)
+        giantMoleBurrow(it)
+    }
+}
+
+fun giantMoleBurrow(mole: NPC) {
+    for (attacker in mole.attackers) {
+        attacker.mode = EmptyMode
+    }
+    mole.attackers.clear()
+    var tileToDust = getTotalDirection(mole.facing, mole.tile)
```

Comment:
I don't think the mechanic for this is correct, I think it picks a random direction. Cause if you're fighting the mole it's always going to be facing you and never going to cover you with mud.

Reply:
okay I will change it to pick a random direction and burrow.

Reply:
Works really well now 👍